ReportingServicesTools/Functions/CatalogItems/Set-RsItemDataSource.ps1

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2017 Microsoft Corporation. All Rights Reserved.
+# Licensed under the MIT License (MIT)
+
+function Set-RsItemDataSource
+{
+    <#
+        .SYNOPSIS
+            This script updates embedded data sources for a report or a shared data set on Report Server.
+        
+        .DESCRIPTION
+            This script updates embedded data sources for a report or a shared data set on Report Server.
+        
+        .PARAMETER RsItem
+            Specify the path to report or shared data set.
+
+        .PARAMETER DataSource
+            Specify the data sources, associated to this catalog item, to update.
+        
+        .PARAMETER ReportServerUri
+            Specify the Report Server URL to your SQL Server Reporting Services Instance.
+            Use the "Connect-RsReportServer" function to set/update a default value.
+        
+        .PARAMETER Credential
+            Specify the credentials to use when connecting to the Report Server.
+            Use the "Connect-RsReportServer" function to set/update a default value.
+        
+        .PARAMETER Proxy
+            Report server proxy to use.
+            Use "New-RsWebServiceProxy" to generate a proxy object for reuse.
+            Useful when repeatedly having to connect to multiple different Report Server.
+
+        .EXAMPLE
+            $dataSources = Get-RsItemDataSource -RsItem '/report'
+            Set-RsItemDataSource -RsItem '/report -DataSource $datasources
```

Comment:
did you miss the ' at the end of the report?

Reply:
Oops. Thanks for pointing this out!